src/UserService.Business/Commands/Image/AddImagesCommand.cs

```diff
@@ -0,0 +1,168 @@
+          ICreateImagesRequest.CreateObj(
+            _createImageDataMapper.Map(request),
+            ImageSource.User));
+
+        if (createResponse.Message.IsSuccess)
+        {
+          return createResponse.Message.Body.ImagesIds;
+        }
+
+        _logger.LogWarning(
+          logMessage + " Errors: {Errors}",
+          string.Join('\n', createResponse.Message.Errors));
+
+        errors.Add(errorMessage);
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, logMessage);
+
+        errors.Add(errorMessage);
+      }
+
+      return null;
+    }
+
+    private Guid? GetUserIdFromEntity(Guid entityId, EntityType entityType)
+    {
+      return entityType switch
+      {
+        EntityType.User => entityId,
+        EntityType.Certificate => _certificateRepository.Get(entityId).UserId,
+        EntityType.Education => _educationRepository.Get(entityId).UserId,
+        _ => null
+      };
+    }
+
+    public AddImagesCommand(
+      IImageRepository imageRepository,
+      ICertificateRepository certificateRepository,
+      IEducationRepository educationRepository,
+      IAccessValidator accessValidator,
+      IAddImagesRequestValidator requestValidator,
+      IDbEntityImageMapper dbEntityImageMapper,
+      ICreateImageDataMapper createImageDataMapper,
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<ICreateImagesRequest> rcCreateImage,
+      ILogger<AddImagesCommand> logger)
+    {
+      _imageRepository = imageRepository;
+      _certificateRepository = certificateRepository;
+      _educationRepository = educationRepository;
+      _accessValidator = accessValidator;
+      _requestValidator = requestValidator;
+      _dbEntityImageMapper = dbEntityImageMapper;
+      _createImageDataMapper = createImageDataMapper;
+      _httpContextAccessor = httpContextAccessor;
+      _rcCreateImage = rcCreateImage;
+      _logger = logger;
+    }
+
+    public async Task<OperationResultResponse<List<Guid>>> Execute(AddImagesRequest request)
+    {
+      OperationResultResponse<List<Guid>> response = new();
+
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+
+      if (!_accessValidator.HasRights(senderId, Rights.AddEditRemoveUsers)
+        && senderId != GetUserIdFromEntity(request.EntityId, request.EntityType))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Not enough rights.");
+
+        return response;
+      }
+
+      if (!_requestValidator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.AddRange(errors);
+
+        return response;
+      }
+
+      List<Guid> result = await AddImages(request.Images, response.Errors);
```

Comment:
you can write response.Body = await AddImages(request.Images, response.Errors); and do not create variable